orderer/consensus/etcdraft/chain.go

```diff
@@ -911,6 +921,41 @@ func (c *Chain) ordered(msg *orderer.SubmitRequest) (batches [][]*common.Envelop
 			}
 		}
 
+		if !c.leaderTransferFailed {
+			if c.checkForEvictionNCertRotation(msg.Payload) {
+				go func() {
+					if transferee, err := c.Node.abdicateLeader(c.raftID); err != nil {
+						if err.Error() == LEADERTRANSFERTIMEOUT.Error() && transferee != raft.None {
+							if transferee != c.leadershipTransferCandidate {
+								c.leadershipTransferCandidate = transferee
+							} else {
+								c.leaderTransferFailed = true
```

Comment:
you're writing to a variable `leaderTransferFailed` in a goroutine without but reading it from the goroutine spawned. This needs to be synchronized.

Reply:
Updated with atomic constructs now.